apps/member/src/components/common/HashtagButton/HashtagButton.tsx

```diff
@@ -0,0 +1,76 @@
+import { useEffect, useState } from 'react';
+
+import { Button } from '@clab-platforms/design-system';
+import { cn } from '@clab-platforms/utils';
+
+import { useHashtag } from '@hooks/queries/hashtag';
+
+import type { Hashtag } from '@type/hashtag';
+
+interface HashtagButtonProps {
+  clicked: Array<string>;
+  onClick: (hashtag: string) => void;
+  className?: string;
+}
+
+const HashtagButton = ({ clicked, onClick, className }: HashtagButtonProps) => {
+  const { data } = useHashtag();
+  const [hashtagCategories, setHashtagCategories] = useState<
+    { title: string; items: Array<Hashtag> }[]
+  >([]);
+
+  useEffect(() => {
+    if (data) {
+      const list = [
+        {
+          title: '언어',
+          items: data.filter((item) => item.hashtagCategory === 'LANGUAGE'),
+        },
+        {
+          title: '분야',
+          items: data.filter((item) => item.hashtagCategory === 'FIELD'),
+        },
+        {
+          title: '기술',
+          items: data.filter((item) => item.hashtagCategory === 'SKILL'),
+        },
+        {
+          title: '기타',
+          items: data.filter((item) => item.hashtagCategory === 'ETC'),
+        },
+      ];
```

Comment:
언어 - LANGUAGE, 분야 - FIELD 이런 데이터들은 매핑이 가능한 데이터들이라서, 상수로 관리했으면 더 깔끔하지 않았나 싶어요. 해당 데이터들을 상수로 관리했더라면 다음과 같이 간단하게 나타낼 수 있었을 듯 해요!

```ts
const HASHTAG = [{ titleKor: '언어', titleEng: 'LANGUAGE' }, ... ];
```

```suggestion
const list = HASHTAG.map(({ titleKor, titleEng }) => ({
  title: titleKor,
  items: data.filter((item) => item.hashtagCategory === titleEng),
}));
```

Reply:
파일들을 확인해보니까 이미 타입이 있더라구요. 
해당 타입을 바로 선언하기보단 상수로 해당 요소들을 작성하고 -> 해당 배열의 요소에 대해서 타입을 생성한다면 유지보수적인 측면에서도 더 좋아질 듯 해요

Reply:
오 좋은 방식이네요! 해당 부분 반영해서 리팩토링 했어요 감사합니다!